onnxruntime/core/session/onnxruntime_c_api.cc

```diff
@@ -755,6 +755,32 @@ ORT_API_STATUS_IMPL(OrtApis::ModelMetadataLookupCustomMetadataMap,
   API_IMPL_END
 }
 
+ORT_API_STATUS_IMPL(OrtApis::ModelMetadataGetCustomMetadataMapKeys,
+                    _In_ const OrtModelMetadata* model_metadata,
+                    _Inout_ OrtAllocator* allocator, _Outptr_ char*** keys, _Out_ int64_t* num_keys) {
+  API_IMPL_BEGIN
+  auto custom_metadata_map =
+      reinterpret_cast<const ::onnxruntime::ModelMetadata*>(model_metadata)->custom_metadata_map;
+
+  auto count = custom_metadata_map.size();
+  if (count == 0) {
+    *keys = nullptr;
+  } else {
+    *keys = reinterpret_cast<char**>(allocator->Alloc(allocator, count * sizeof(char*)));
```

Comment:
Use SafeInt when calling Alloc. There are plenty of such examples in the codebase. Also see https://github.com/microsoft/onnxruntime/blob/master/docs/Coding_Conventions_and_Standards.md.

Reply:
Done, thanks!